stv/stv.py

```diff
@@ -427,34 +427,34 @@ def main(cmd=None):
     ballots = []
     ballots_file = sys.stdin
     if args.ballots_file != 'sys.stdin':
-        ballots_file = open(args.ballots_file, 'U')
-    ballots_reader = csv.reader(ballots_file, delimiter=',',
-                                quotechar='"',
-                                skipinitialspace=True)
-    for ballot in ballots_reader:
-        ballots.append(Ballot(ballot))
-
-    if args.seats == 0:
-        args.seats = len(ballots) // 2
-
-    constituencies = {}
-    if args.constituencies_file:
-        constituencies_file = open(args.constituencies_file, 'U')
-        constituencies_reader = csv.reader(constituencies_file,
-                                           delimiter=',',
-                                           quotechar='"',
-                                           skipinitialspace=True)
-        constituency_id = 0
-        for constituency in constituencies_reader:
-            for candidate in constituency:
-                constituencies[candidate] = constituency_id
-            constituency_id += 1
-
-    (elected, vote_count, full_data) = count_stv(ballots, args.seats, args.droop,
-                                      constituencies,
-                                      args.quota,
-                                      args.random,
-                                      logger=logger)
+        with open(args.ballots_file) as f:
+            ballots_reader = csv.reader(f, delimiter=',',
```

Comment:
Change in the indentation starting in this line seems to change behaviour of this code.

It looks like previously we could've read ballots either from file or `sys.stdin`, but now I'm not sure if the stdin way still will work.

Reply:
+1. In this case a refactoring might be a bit more complicated, you could leave it as it was before for now.